### PR TITLE
feat: reconfigure tqdm progress bar in %%bigquery magic

### LINKED ISSUE #1378: module 'tqdm' has no attribute 'notebook' error
#### Environment details

  - OS type and version: 
      - google colab 
      - Linux 224200c9dd15 5.10.133+ #1 SMP Fri Aug 26 08:44:51 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux
      - PRETTY_NAME="Ubuntu 18.04.6 LTS"
  - Python version: `python --version`
      - Python 3.7.14
  - pip version: `pip --version`
      - pip 21.1.3 from /usr/local/lib/python3.7/dist-packages/pip (python 3.7) 
  - `google-cloud-bigquery` version: `pip show google-cloud-bigquery`
      - 3.3.3

#### Steps to reproduce

1. run this code  cell 1 at google colab.
2. restart google colab runtime.
3. run this code cell 1 - all
4. occur error

1-alt: if Cell 1-alt instead of cell 1, no error.

#### Code example

Cell 1
```python

query = """
SELECT
  name, gender,
  SUM(number) AS total
FROM
  `bigquery-public-data.usa_names.usa_1910_2013`
GROUP BY
  name, gender
ORDER BY
  total DESC
LIMIT
  10
"""
%pip install --upgrade google-cloud-bigquery db-dtypes
```

Cell 2
```python

from google.cloud import bigquery

from google.colab import auth
import pandas as pd

auth.authenticate_user()
project_id="<your project name>"
pd.read_gbq(query, project_id=project_id, progress_bar_type="tqdm_notebook")

```

cell 1-alt

```python

query = """
SELECT
  name, gender,
  SUM(number) AS total
FROM
  `bigquery-public-data.usa_names.usa_1910_2013`
GROUP BY
  name, gender
ORDER BY
  total DESC
LIMIT
  10
"""
#%pip install --upgrade google-cloud-bigquery db-dtypes

```

#### Stack trace
```
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
[<ipython-input-3-e0e6ef23fa30>](https://localhost:8080/#) in <module>
      1 auth.authenticate_user()
      2 project_id="<your project name>"
----> 3 pd.read_gbq(query, project_id=project_id, progress_bar_type="tqdm_notebook")

6 frames
[/usr/local/lib/python3.7/dist-packages/pandas/io/gbq.py](https://localhost:8080/#) in read_gbq(query, project_id, index_col, col_order, reauth, auth_local_webserver, dialect, location, configuration, credentials, use_bqstorage_api, max_results, progress_bar_type)
    192         configuration=configuration,
    193         credentials=credentials,
--> 194         **kwargs,
    195     )
    196 

[/usr/local/lib/python3.7/dist-packages/pandas_gbq/gbq.py](https://localhost:8080/#) in read_gbq(query, project_id, index_col, col_order, reauth, auth_local_webserver, dialect, location, configuration, credentials, use_bqstorage_api, max_results, verbose, private_key, progress_bar_type)
    965         configuration=configuration,
    966         max_results=max_results,
--> 967         progress_bar_type=progress_bar_type,
    968     )
    969 

[/usr/local/lib/python3.7/dist-packages/pandas_gbq/gbq.py](https://localhost:8080/#) in run_query(self, query, max_results, progress_bar_type, **kwargs)
    530             query_reply,
    531             max_results=max_results,
--> 532             progress_bar_type=progress_bar_type,
    533         )
    534 

[/usr/local/lib/python3.7/dist-packages/pandas_gbq/gbq.py](https://localhost:8080/#) in _download_results(self, query_job, max_results, progress_bar_type)
    560                 dtypes=nullsafe_dtypes,
    561                 bqstorage_client=bqstorage_client,
--> 562                 progress_bar_type=progress_bar_type,
    563             )
    564         except self.http_error as ex:

[/usr/local/lib/python3.7/dist-packages/google/cloud/bigquery/table.py](https://localhost:8080/#) in to_dataframe(self, bqstorage_client, dtypes, progress_bar_type, create_bqstorage_client, geography_as_object)
   1980             progress_bar_type=progress_bar_type,
   1981             bqstorage_client=bqstorage_client,
-> 1982             create_bqstorage_client=create_bqstorage_client,
   1983         )
   1984 

[/usr/local/lib/python3.7/dist-packages/google/cloud/bigquery/table.py](https://localhost:8080/#) in to_arrow(self, progress_bar_type, bqstorage_client, create_bqstorage_client)
   1776         try:
   1777             progress_bar = get_progress_bar(
-> 1778                 progress_bar_type, "Downloading", self.total_rows, "rows"
   1779             )
   1780 

[/usr/local/lib/python3.7/dist-packages/google/cloud/bigquery/_tqdm_helpers.py](https://localhost:8080/#) in get_progress_bar(progress_bar_type, description, total, unit)
     50             return tqdm.tqdm(desc=description, total=total, unit=unit)
     51         elif progress_bar_type == "tqdm_notebook":
---> 52             return tqdm.notebook.tqdm(desc=description, total=total, unit=unit)
     53         elif progress_bar_type == "tqdm_gui":
     54             return tqdm.tqdm_gui(desc=description, total=total, unit=unit)

AttributeError: module 'tqdm' has no attribute 'notebook'
```

### PR DESCRIPTION
**Pull request by aribray**
Change-Id: I2add62e3cdd5f25f88ace2d08f212796918158b6

Adds the BigQuery job ID to the `tqdm progress bar description` for a more helpful user message. Adds the correct `tqdm.notebook` import. Changes the default progress bar to `tqdm_notebook`

fixes: #1378 
See internal 242869795

### CODE REVIEW

File: google/cloud/bigquery/_tqdm_helpers.py [hunk 1]
```diff
@@ -106,7 +106,7 @@ def wait_for_query(
             )
             progress_bar.update(default_total)
             progress_bar.set_description(
-                "Query complete after {:0.2f}s".format(time.time() - start_time),
+                "BigQuery job ID {} successfully executed".format(query_job.job_id),
```
Comment (chalmerlowe) [inline google/cloud/bigquery/_tqdm_helpers.py:109]: Why remove the duration as opposed to combining the duration and job id?
Also, f-strings are often better than the format method for combining strings and variable values. I recommend them highly.


```suggestion
                f"BigQuery job ID {query_job.job_id} successfully executed in {time.time() - start_time:0.2f} seconds.",
```

File: google/cloud/bigquery/magics/magics.py [hunk 2]
```diff
@@ -338,13 +338,21 @@ def _run_query(client, query, job_config=None):
...
+        )
         try:
             query_job.result(timeout=0.5)
             break
         except futures.TimeoutError:
             continue
-    print("\nQuery complete after {:0.2f}s".format(time.time() - start_time))
+    print(
+        "\nJob ID {} successfully executed after {:0.2f}s".format(
+            query_job.job_id, time.perf_counter() - start_time
```
Comment (chalmerlowe) [inline google/cloud/bigquery/magics/magics.py:352]: I had not used perf_counter() before. Looks very useful.

File: google/cloud/bigquery/table.py [hunk 3]
```diff
@@ -2077,7 +2077,7 @@ def to_geodataframe(
                   Use the :func:`tqdm.tqdm` function to print a progress bar
                   to :data:`sys.stderr`.
                 ``'tqdm_notebook'``
```
Comment (leahecole) [inline google/cloud/bigquery/table.py:2079]: does this line also need to change? (it might not but I just noticed it stayed the same)

Comment (aribray) [reply google/cloud/bigquery/table.py:2079]: Yep, it should have gotten changed with the function changed, but I'll update it for this PR!

File: google/cloud/bigquery/_tqdm_helpers.py (hunk 1 above)
Comment (aribray) [reply google/cloud/bigquery/_tqdm_helpers.py]: The duration isn't being calculated correctly, so I removed it. We can reopen a separate issue to dig into that more.

File: google/cloud/bigquery/_tqdm_helpers.py [hunk 4]
```diff
@@ -15,13 +15,15 @@
...
 import concurrent.futures
+import sys
 import time
 import typing
 from typing import Optional
 import warnings
 
 try:
     import tqdm  # type: ignore
+    import tqdm.notebook as notebook  # type: ignore
```
Comment (shollyman) [inline google/cloud/bigquery/_tqdm_helpers.py:26]: This additional import test is to ensure we're dealing with a modern tqdm installation?  We don't test for tqdm_gui, should we?  Do we use it in a meaningful way?

**Review by shollyman — APPROVED** [review]
One minor question about the helpers, otherwise LGTM.

File: google/cloud/bigquery/_tqdm_helpers.py (hunk 4 above)
Comment (aribray) [reply google/cloud/bigquery/_tqdm_helpers.py:26]: Right, line 26 is the correct import

File: google/cloud/bigquery/_tqdm_helpers.py [hunk 5]
```diff
@@ -95,19 +110,15 @@ def wait_for_query(
...
+                f"Query executing stage {current_stage.name} and status {current_stage.status} : {time.perf_counter() - start_time:.2f}s"
             )
         try:
             query_result = query_job.result(
                 timeout=_PROGRESS_BAR_UPDATE_INTERVAL, max_results=max_results
             )
             progress_bar.update(default_total)
             progress_bar.set_description(
-                "Query complete after {:0.2f}s".format(time.time() - start_time),
+                f"Job ID {query_job.job_id} successfully executed",
```
Comment (tswast) [inline google/cloud/bigquery/_tqdm_helpers.py:121]: Might want to include the location in the output too, formatted like the bq cli as location:I'd https://cloud.google.com/python/docs/reference/bigquery/latest/google.cloud.bigquery.job.QueryJob#google_cloud_bigquery_job_QueryJob_location

I believe the location is required to fetch the job metadata for the non-default location.

**Review by tswast — COMMENTED** [review]
Looks great! Thanks so much for working through those testing issues. Just one nit, that might be worth a little follow up

File: google/cloud/bigquery/magics/magics.py [hunk 6]
```diff
@@ -286,7 +286,7 @@ def _handle_error(error, destination_var=None):
 
     Args:
         error (Exception):
-            An exception that ocurred during the query exectution.
+            An exception that ocurred during the query execution.
```
Comment (chalmerlowe) [inline google/cloud/bigquery/magics/magics.py:289]: Change the word `ocurred` to the correct spelling of `occurred`

`An exception that occurred during the query execution.`

File: tests/system/test_magics.py [hunk 7]
```diff
@@ -71,8 +71,7 @@ def test_bigquery_magic(ipython_interactive):
     # Removes blanks & terminal code (result of display clearing)
     updates = list(filter(lambda x: bool(x) and x != "\x1b[2K", lines))
     assert re.match("Executing query with job ID: .*", updates[0])
-    assert all(re.match("Query executing: .*s", line) for line in updates[1:-1])
-    assert re.match("Query complete after .*s", updates[-1])
+    assert (re.match("Query executing: .*s", line) for line in updates[1:-1])
```
Comment (chalmerlowe) [inline tests/system/test_magics.py:74]: Using `assert` with a `tuple` will **always** return `True`, no matter what is in the `tuple` unless the `tuple` is **empty**.
Are we trying to detect whether any matches are True OR are we trying to detect whether **ALL** the responses are True (...as implied by the deleted code above)?

```
In [9]: assert (True, True, True)
<>:1: SyntaxWarning: assertion is always true, perhaps remove parentheses?
<ipython-input-9-0a5188c9e554>:1: SyntaxWarning: assertion is always true, perhaps remove parentheses?
  assert (True, True, True)

In [10]: assert (False, False, False)
<>:1: SyntaxWarning: assertion is always true, perhaps remove parentheses?
<ipython-input-10-f31b4c57c272>:1: SyntaxWarning: assertion is always true, perhaps remove parentheses?
  assert (False, False, False)

In [11]: assert (False, False, True)
<>:1: SyntaxWarning: assertion is always true, perhaps remove parentheses?
<ipython-input-11-59e8f7bc6fa4>:1: SyntaxWarning: assertion is always true, perhaps remove parentheses?
  assert (False, False, True)

In [14]: assert (True,)
<>:1: SyntaxWarning: assertion is always true, perhaps remove parentheses?
<ipython-input-14-7b46914816ee>:1: SyntaxWarning: assertion is always true, perhaps remove parentheses?
  assert (True,)

In [15]: assert (False,)
<>:1: SyntaxWarning: assertion is always true, perhaps remove parentheses?
<ipython-input-15-a62a5cb79bea>:1: SyntaxWarning: assertion is always true, perhaps remove parentheses?
  assert (False,)
  
  In [16]: assert ()
---------------------------------------------------------------------------
AssertionError                            Traceback (most recent call last)
Input In [16], in <cell line: 1>()
----> 1 assert ()
```

**Review by chalmerlowe — COMMENTED** [review]
A couple of minor tweaks and a question.
Other than that, looks good to me.

**Pull request merged**